crates/mun_hir/src/ty/snapshots/tests__comparison_not_implemented_for_struct.snap

```diff
@@ -0,0 +1,9 @@
+---
+source: crates/mun_hir/src/ty/tests.rs
+expression: "struct Foo;\n\nfn main() -> bool {\n    Foo == Foo\n}"
+---
+[37; 47): cannot apply binary operator
+[31; 49) '{     ... Foo }': bool
+[37; 40) 'Foo': Foo
+[37; 47) 'Foo == Foo': bool
```

Comment:
It's interesting that it still knows what type inference results in, even though it is *illegal*

Reply:
That because applying the equality operator will always have a boolean as return type no matter the left- and right-hand types.

Reply:
Gotcha